rust/agama-dbus-server/src/questions.rs

```diff
@@ -58,22 +69,19 @@ impl LuksQuestionObject {
     pub fn set_password(&mut self, value: &str) -> () {
         self.0.password = value.to_string();
     }
-
-    #[dbus_interface(property)]
-    pub fn attempt(&self) -> u8 {
-        self.0.attempt
-    }
 }
 
 /// Question types used to be able to properly remove object from dbus
 enum QuestionType {
-    Generic,
-    Luks,
+    Base,
+    BaseWithPassword,
 }
 
 trait AnswerStrategy {
-    /// TODO: find way to be able to answer any type of question, not just generic
+    /// Provides answer for generic question
     fn answer(&self, question: &GenericQuestion) -> Option<String>;
+    /// Provides answer and password for base question with password
+    fn answer_with_password(&self, question: &WithPassword) -> (Option<String>, Option<String>);
```

Comment:
again I don't understand why the returns are optional

Also, is their optionality independent?

Reply:
yes and yes. None means that strategy does not provide it and it is up to question to decide what it means. E.g. Default strategy for password should return something like `(Some(self.default_answer), None)` which means use default answer and does not provide password.